### PR TITLE
sidevm: auto restart crashed instances

### PR DESCRIPTION
**Pull request by kvinwang**

### CODE REVIEW

File: crates/pink/sidevm/host-runtime/src/service.rs [hunk 1]
```diff
@@ -15,14 +16,22 @@ pub enum Report {
     VmTerminated { id: VmId, reason: ExitReason },
 }
 
-#[derive(Debug)]
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
 pub enum ExitReason {
+    // The program returned from main
```
Comment (h4x3rotab) [inline crates/pink/sidevm/host-runtime/src/service.rs:21]: Good to make it docstring

File: crates/phactory/src/contracts/support.rs [hunk 2]
```diff
@@ -258,24 +270,44 @@ impl FatContract {
...
-                let handle = do_start_sidevm(
+            let guard = sidevm_info.handle.lock().unwrap();
+            let handle = if let SidevmHandle::Terminated(reason) = &*guard {
+                let need_restart = match reason {
+                    ExitReason::Exited(_) => false,
+                    ExitReason::Stopped => false,
+                    ExitReason::InputClosed => false,
+                    ExitReason::Panicked => true,
+                    ExitReason::Cancelled => false,
+                    // TODO.kevin: Allow to charge new gas? How to charge gas or weather the gas
```
Comment (h4x3rotab) [inline crates/phactory/src/contracts/support.rs:291]: Let's discuss it offline

File: crates/pink/sidevm/host-runtime/src/service.rs (hunk 1 above)
Comment (kvinwang) [reply crates/pink/sidevm/host-runtime/src/service.rs]: Done

**Review by h4x3rotab — APPROVED** [review]
LGTM

**Pull request merged**